substrate-node/pallets/pallet-smart-contract/src/lib.rs

```diff
@@ -846,52 +912,117 @@ impl<T: Config> Pallet<T> {
         ContractBillingInformationByID::<T>::insert(report.contract_id, &contract_billing_info);
     }
 
-    pub fn _bill_contracts_at_block(block: T::BlockNumber) -> DispatchResultWithPostInfo {
-        let current_block_u64: u64 = block.saturated_into::<u64>();
-        let contracts = ContractsToBillAt::<T>::get(current_block_u64);
-        for contract_id in contracts {
-            if !Contracts::<T>::contains_key(contract_id) {
-                continue;
+    pub fn _bill_contract_for_block(
+        contract_id: u64,
+        block: T::BlockNumber,
+    ) -> DispatchResultWithPostInfo {
+        if !Contracts::<T>::contains_key(contract_id) {
+            return Ok(().into());
+        }
+        let mut contract = Contracts::<T>::get(contract_id).ok_or(Error::<T>::ContractNotExists)?;
```

Comment:
try so see if we can remove `contains_key` and handle the result in the `get` call on line 922